src/components/Capture/withCameraDetection.tsx

```diff
@@ -10,49 +10,40 @@ type Props = ReduxProps & WithCameraDetectionProps
 export default function withCameraDetection<P extends Props>(
   WrappedComponent: ComponentType<P>
 ): ComponentType<P> {
-  return class WithCameraDetection extends PureComponent<P> {
-    state: WithCameraDetectionProps = {
-      hasCamera: null,
-    }
+  const WithCameraDetection: FunctionComponent<P> = (props) => {
+    // FIXME: hasCamera should use the type in WithCameraDetectionProps
+    const [hasCamera, setHasCamera] = useState<boolean | null>(null)
+    const [cameraCheckerIntervalId, setCameraCheckerIntervalId] = useState<
+      number | null
+    >(null)
+
+    useEffect(() => {
+      const checkCameraSupport = () =>
+        checkIfHasWebcam((cameraState: boolean) => {
+          if (hasCamera !== cameraState) {
+            props.actions.setDeviceHasCameraSupport(cameraState)
+            setHasCamera(cameraState)
+          }
+        })
 
-    private cameraCheckerIntervalId?: number
-
-    componentDidMount() {
       // HACK: use of isSafari131 util function is a workaround specifically for
       //       Safari 13.1 bug that incorrectly returns a "videoinput" as "audioinput"
       //       on subsequent calls to enumerateDevices
       //       see https://bugs.webkit.org/show_bug.cgi?id=209580
       if (!isSafari131()) {
```

Comment:
Out of scope: this wasn't covered by unit tests. I'd be great if you can add some tests to cover it (please ignore it if you need this PR to be merged quickly)
![swappy-20210416-110046](https://user-images.githubusercontent.com/1215541/115008560-299dd700-9ea3-11eb-8294-effab6aa9c1c.png)

Reply:
Agree with above point, but we should create a ticket to make sure it's done if it won't be covered in this PR.

Reply:
I agree, will create a ticket as I believe we want to start the patch release process today